trinity/sync/full/chain.py

```diff
@@ -85,314 +143,194 @@ class FastChainSyncer(BaseHeaderChainSyncer):
         commands.Transactions,
         commands.NewBlockHashes,
     }
+    _pending_bodies: Dict[BlockHeader, BlockBody]
+
+    def __init__(self,
+                 chain: AsyncChain,
+                 db: AsyncHeaderDB,
+                 peer_pool: PeerPool,
+                 token: CancelToken = None) -> None:
+        super().__init__(chain, db, peer_pool, token)
+        self._pending_bodies = {}
+
+        # queue up any idle peers, in order of how fast they return block bodies
+        self._body_peers = WaitingPeers(commands.BlockBodies)
+
+        # Track incomplete block body download tasks
+        # - arbitrarily allow up to 4 requests-worth of headers queued up
+        # - try to get bodies from lower block numbers first
+        self._block_body_tasks = TaskQueue(MAX_BODIES_FETCH * 4, attrgetter('block_number'))
+
+    # TODO move this to BaseService if it gets broader usage and/or stays stable for long enough
+    def delayed_run_func(self, func: Callable[[], None], delay: float) -> None:
```

Comment:
This seems like it should be using [`loop.call_later()`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.AbstractEventLoop.call_later) or `call_at`